pkg/controllers/nodeclaim/disruption/drift.go

```diff
@@ -81,6 +84,21 @@ func (d *Drift) isDrifted(ctx context.Context, nodePool *v1.NodePool, nodeClaim
 	}); reason != "" {
 		return reason, nil
 	}
+	// Mark the NodeClaim as drifted due to a stale instance type if
+	// 1. The NodeClaim doesn't have the instance type label
+	// 2. The NodeClaim has an instance type that doesn't exist in the cloudprovider instance types
+	// 3. There are no offerings that match the requirements
+	its, err := d.cloudProvider.GetInstanceTypes(ctx, nodePool)
+	if err != nil {
+		return "", err
+	}
+	it, ok := lo.Find(its, func(it *cloudprovider.InstanceType) bool {
+		return it.Name == nodeClaim.Labels[corev1.LabelInstanceTypeStable]
+	})
+	if !ok || len(it.Offerings.Compatible(scheduling.NewLabelRequirements(nodeClaim.Labels))) == 0 {
```

Comment:
Does the offerings compatibility in the NodeClaim compatibility? I forget if these are scoped together or not?